### PR TITLE
Update Storybook to 6.3.x

### PR DESCRIPTION
**Pull request by jcq**
### Description
This updates Storybook and related dependencies to 6.3.x (latest). It also updates a number of stories to remove use of deprecated `knobs` addon.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -61,14 +61,15 @@
...
-    "@storybook/react": "^6.1.17",
+    "@storybook/addon-a11y": "^6.3.8",
+    "@storybook/addon-actions": "^6.3.8",
+    "@storybook/addon-controls": "^6.3.8",
+    "@storybook/addon-docs": "^6.3.8",
+    "@storybook/addon-knobs": "^6.3.1",
+    "@storybook/addon-links": "^6.3.8",
+    "@storybook/addons": "^6.3.8",
+    "@storybook/react": "^6.3.8",
+    "@storybook/testing-react": "^0.0.22",
```
Comment (jcq) [inline client/package.json:72]: This package contains utility functions for utilizing Storybook stories within our Jest tests. For more details, see blog post:
https://storybook.js.org/blog/stories-are-tests/

File: client/stories/CaseflowLogo.stories.js [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+    }
+  }
 };
 
-export const all = () => (
+const Template = (args) => <CaseflowLogo {...args} />;
+
+export const All = () => (
   <React.Fragment>
     <CaseflowLogo accentColor="#5c626b" overlapColor="#4d535b" />
```
Comment (meckila) [inline client/stories/CaseflowLogo.stories.js:30]: Can this one use a default color constant like so? 
```
accentColor={LOGO_COLORS.DEFAULT}
```

**Review by meckila — APPROVED** [review]
Approved with non-blocking comments

**Comment by meckila** [comment]
👍🏽 for replacing knobs with controls
Intrigued by @storybook/testing-react and how useful it will be to our current jest tests

File: client/stories/CaseflowLogo.stories.js (hunk 2 above)
Comment (jcq) [reply client/stories/CaseflowLogo.stories.js]: Not sure why folks had chosen not to do so in the past, but yeah — makes sense to me. 😁

**Pull request merged**